model-optimizer/extensions/middle/AddMeanScaleValues.py

```diff
@@ -91,28 +91,27 @@ def find_and_replace_pattern(self, graph: Graph):
         input_nodes = graph.get_op_nodes(op='Parameter')
 
         if not isinstance(values, dict):
+            # The case when input names to apply mean/scales weren't specified
             if len(values) != len(input_nodes):
                 raise Error('Numbers of inputs and mean/scale values do not match. ' + refer_to_faq_msg(61))
 
             data = np.copy(values)
             values = {}
             for idx, node in enumerate(input_nodes):
-                assert node.has_valid('name')
                 values.update(
                     {
-                        node['name']: {
+                        node.soft_get('name', node.id): {
                             'mean': data[idx][0],
                             'scale': data[idx][1]
                         }
                     }
                 )
 
-        for node_name in values:
-            node_mean_scale_values = values[node_name]
-            node_name, port = split_node_in_port(node_name)
+        for node_name, node_mean_scale_values in values.items():
             node_id = None
             try:
-                node_id = graph.get_node_id_by_name(node_name)
+                node_id, direction, port = get_node_id_with_ports(graph, node_name)
+                assert direction != 'out', 'Only input port can be specified for mean/scale application'
```

Comment:
nit: may be it is a bit easier to understand if the assert will be like this `assert direction == 'in'`

Reply:
Direction can be None if no port provided and it is valid for this code.